sql-statements/sql-statement-overview.md

```diff
@@ -0,0 +1,3 @@
+# SQL Statement Overview
+
+TiDB uses SQL statements that try to follow the ISO/IEC SQL standards and are extended with MySQL and TiDB specific statements where required.
```

Comment:
is this file expected to only contain one sentence? will this be expanded later?

Reply:
I added this to have one place to link to instead of the old sqlgram url. 

We could list all the statements on this page, but on the other hand, they are already in the TOC.

We could add more generic information about statements here, but I don't know what to add.

We could remove this, but then what place could we link to instead of the old sqlgram url?

@kennytm @qiancai Any ideas?

Reply:
How about listing all the statements on this overview page according to the statement categories? Then users can view the statements in two dimensions, one in alphabetical order in the left navigation pane, and the other in the category order on this overview page. 

Here is an example (it is generated by AI according to the TiDB toc). Please feel free to adjust it if necessary. 

```
## Data definition statements (DDL)

| SQL Statement | Description |
|---------------|-------------|
| [`ADD COLUMN`](/sql-statements/sql-statement-add-column.md) | Adds a column to an existing table. |
| [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) | Adds an index to an existing table. |
| [`ADD PLACEMENT POLICY`](/sql-statements/sql-statement-add-placement-policy.md) | Adds a placement policy to the database. |
| [`ADD PRIMARY KEY`](/sql-statements/sql-statement-add-primary-key.md) | Adds a primary key to an existing table. |
| [`ADD RESOURCE LABEL`](/sql-statements/sql-statement-add-resource-label.md) | Adds a resource label to the system. |
| [`ADD RESOURCE GROUP`](/sql-statements/sql-statement-add-resource-group.md) | Adds a resource group to the system. |
| [`ADD STATS`](/sql-statements/sql-statement-add-stats.md) | Adds statistics to a table. |
| [`ANALYZE`](/sql-statements/sql-statement-analyze.md) | Collects statistics about a table. |
| [`CHANGE COLUMN`](/sql-statements/sql-statement-change-column.md) | Changes a column definition. |
| [`CREATE DATABASE`](/sql-statements/sql-statement-create-database.md) | Creates a new database. |
| [`CREATE INDEX`](/sql-statements/sql-statement-create-index.md) | Creates a new index on a table. |
| [`CREATE PLACEMENT POLICY`](/sql-statements/sql-statement-create-placement-policy.md) | Creates a new placement policy. |
| [`CREATE RESOURCE GROUP`](/sql-statements/sql-statement-create-resource-group.md) | Creates a new resource group. |
| [`CREATE SEQUENCE`](/sql-statements/sql-statement-create-sequence.md) | Creates a new sequence object. |
| [`CREATE TABLE`](/sql-statements/sql-statement-create-table.md) | Creates a new table. |
| [`CREATE USER`](/sql-statements/sql-statement-create-user.md) | Creates a new user. |
| [`CREATE VIEW`](/sql-statements/sql-statement-create-view.md) | Creates a new view. |
| [`DROP COLUMN`](/sql-statements/sql-statement-drop-column.md) | Drops a column from a table. |
| [`DROP DATABASE`](/sql-statements/sql-statement-drop-database.md) | Drops an existing database. |
| [`DROP INDEX`](/sql-statements/sql-statement-drop-index.md) | Drops an index from a table. |
| [`DROP PLACEMENT POLICY`](/sql-statements/sql-statement-drop-placement-policy.md) | Drops an existing placement policy. |
| [`DROP RESOURCE GROUP`](/sql-statements/sql-statement-drop-resource-group.md) | Drops a resource group. |
| [`DROP ROLE`](/sql-statements/sql-statement-drop-role.md) | Drops an existing role. |
| [`DROP SEQUENCE`](/sql-statements/sql-statement-drop-sequence.md) | Drops a sequence object. |
| [`DROP STATS`](/sql-statements/sql-statement-drop-stats.md) | Drops statistics from a table. |
| [`DROP TABLE`](/sql-statements/sql-statement-drop-table.md) | Drops an existing table. |
| [`DROP USER`](/sql-statements/sql-statement-drop-user.md) | Drops an existing user. |
| [`DROP VIEW`](/sql-statements/sql-statement-drop-view.md) | Drops an existing view. |
| [`MODIFY COLUMN`](/sql-statements/sql-statement-modify-column.md) | Modifies a column definition. |
| [`RENAME USER`](/sql-statements/sql-statement-rename-user.md) | Renames an existing user. |
| [`RENAME INDEX`](/sql-statements/sql-statement-rename-index.md) | Renames an index. |
| [`RENAME TABLE`](/sql-statements/sql-statement-rename-table.md) | Renames a table. |
| [`TRUNCATE`](/sql-statements/sql-statement-truncate.md) | Truncates all data from a table. |

## Data manipulation statements (DML)

| SQL Statement | Description |
|---------------|-------------|
| [`DELETE`](/sql-statements/sql-statement-delete.md) | Deletes rows from a table. |
| [`INSERT`](/sql-statements/sql-statement-insert.md) | Inserts new rows into a table. |
| [`REPLACE`](/sql-statements/sql-statement-replace.md) | Replaces existing rows or inserts new rows. |
| [`SELECT`](/sql-statements/sql-statement-select.md) | Retrieves rows from a table. |
| [`UPDATE`](/sql-statements/sql-statement-update.md) | Updates existing rows in a table. |
| [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md) | Analyzes and collects statistics for a table. |
| [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md) | Imports data into a table from an external source. |
| [`LOAD DATA`](/sql-statements/sql-statement-load-data.md) | Loads data into a table from a file. |

## Transaction statements

| SQL Statement | Description |
|---------------|-------------|
| [`BEGIN`](/sql-statements/sql-statement-begin.md) | Begins a new transaction. |
| [`COMMIT`](/sql-statements/sql-statement-commit.md) | Commits the current transaction. |
| [`ROLLBACK`](/sql-statements/sql-statement-rollback.md) | Rolls back the current transaction. |
| [`SAVEPOINT`](/sql-statements/sql-statement-savepoint.md) | Sets a savepoint within a transaction. |
| [`START TRANSACTION`](/sql-statements/sql-statement-start-transaction.md) | Starts a new transaction. |
| [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) | Sets the characteristics of the current transaction. |

## Administrative statements

| SQL Statement | Description |
|---------------|-------------|
| [`ADMIN`](/sql-statements/sql-statement-admin.md) | Performs various administrative tasks. |
| [`ADMIN CANCEL DDL`](/sql-statements/sql-statement-admin-cancel-ddl.md) | Cancels a DDL job. |
| [`ADMIN CHECKSUM TABLE`](/sql-statements/sql-statement-admin-checksum-table.md) | Computes and checks the checksum of a table. |
| [`ADMIN CHECK [TABLE\|INDEX]`](/sql-statements/sql-statement-admin-check-table-index.md) | Checks the integrity of a table or index. |
| [`ADMIN CLEANUP`](/sql-statements/sql-statement-admin-cleanup.md) | Cleans up residual files or resources. |
| [`ADMIN PAUSE DDL`](/sql-statements/sql-statement-admin-pause-ddl.md) | Pauses DDL operations. |
| [`ADMIN RECOVER INDEX`](/sql-statements/sql-statement-admin-recover.md) | Recovers a damaged index. |
| [`ADMIN RESUME DDL`](/sql-statements/sql-statement-admin-resume-ddl.md) | Resumes DDL operations. |
| [`ADMIN [SET\|SHOW\|UNSET] BDR ROLE`](/sql-statements/sql-statement-admin-bdr-role.md) | Manages BDR roles. |
| [`ADMIN SHOW DDL [JOBS\|JOB QUERIES]`](/sql-statements/sql-statement-admin-show-ddl.md) | Shows DDL jobs or job queries. |
| [`ADMIN SHOW TELEMETRY`](/sql-statements/sql-statement-admin-show-telemetry.md) | Shows telemetry data. |
| [`BACKUP`](/sql-statements/sql-statement-backup.md) | Creates a backup of the database. |
| [`CANCEL IMPORT JOB`](/sql-statements/sql-statement-cancel-import-job.md) | Cancels an ongoing import job. |
| [`CALIBRATE RESOURCE`](/sql-statements/sql-statement-calibrate-resource.md) | Calibrates system resources. |
| [`FLASHBACK CLUSTER`](/sql-statements/sql-statement-flashback-cluster.md) | Restores the cluster to a previous state. |
| [`FLASHBACK DATABASE`](/sql-statements/sql-statement-flashback-database.md) | Restores the database to a previous state. |
| [`FLASHBACK TABLE`](/sql-statements/sql-statement-flashback-table.md) | Restores a table to a previous state. |
| [`RESTORE`](/sql-statements/sql-statement-restore.md) | Restores a database from a backup. |
| [`SHUTDOWN`](/sql-statements/sql-statement-shutdown.md) | Shuts down the database server. |
| [`SPLIT REGION`](/sql-statements/sql-statement-split-region.md) | Splits a region into smaller regions. |

## Prepared statements

| SQL Statement | Description |
|---------------|-------------|
| [`DEALLOCATE`](/sql-statements/sql-statement-deallocate.md) | Deallocates a previously prepared statement. |
| [`EXECUTE`](/sql-statements/sql-statement-execute.md) | Executes a previously prepared statement. |
| [`PREPARE`](/sql-statements/sql-statement-prepare.md) | Prepares a statement for execution. |

## Utility statements

| SQL Statement | Description |
|---------------|-------------|
| [`DESC`](/sql-statements/sql-statement-desc.md) | Describes the structure of a table. |
| [`DESCRIBE`](/sql-statements/sql-statement-describe.md) | Describes the structure of a table. |
| [`DO`](/sql-statements/sql-statement-do.md) | Executes an expression. |
| [`EXPLAIN`](/sql-statements/sql-statement-explain.md) | Shows the execution plan of a query. |
| [`HELP`](/sql-statements/sql-statement-help.md) | Provides help information. |
| [`USE`](/sql-statements/sql-statement-use.md) | Sets the current database. |

## Show statements

| SQL Statement | Description |
|---------------|-------------|
| [`SHOW`](/sql-statements/sql-statement-show.md) | Shows various information about the database. |
| [`SHOW CREATE DATABASE`](/sql-statements/sql-statement-show-create-database.md) | Shows the CREATE statement for a database. |
| [`SHOW CREATE PLACEMENT POLICY`](/sql-statements/sql-statement-show-create-placement-policy.md) | Shows the CREATE statement for a placement policy. |
| [`SHOW CREATE RESOURCE GROUP`](/sql-statements/sql-statement-show-create-resource-group.md) | Shows the CREATE statement for a resource group. |
| [`SHOW CREATE SEQUENCE`](/sql-statements/sql-statement-show-create-sequence.md) | Shows the CREATE statement for a sequence. |
| [`SHOW CREATE TABLE`](/sql-statements/sql-statement-show-create-table.md) | Shows the CREATE statement for a table. |
| [`SHOW CREATE USER`](/sql-statements/sql-statement-show-create-user.md) | Shows the CREATE statement for a user. |
| [`SHOW DATABASES`](/sql-statements/sql-statement-show-databases.md) | Shows all databases. |
| [`SHOW DRAINER STATUS`](/sql-statements/sql-statement-show-drainer-status.md) | Shows the status of the drainer. |
| [`SHOW ENGINES`](/sql-statements/sql-statement-show-engines.md) | Shows available storage engines. |
| [`SHOW ERRORS`](/sql-statements/sql-statement-show-errors.md) | Shows error messages. |
| [`SHOW FIELDS FROM`](/sql-statements/sql-statement-show-fields-from.md) | Shows columns of a table. |
| [`SHOW GRANTS`](/sql-statements/sql-statement-show-grants.md) | Shows grants for a user. |
| [`SHOW IMPORT JOB`](/sql-statements/sql-statement-show-import-job.md) | Shows the status of an import job. |
| [`SHOW INDEXES`](/sql-statements/sql-statement-show-indexes.md) | Shows indexes of a table. |
| [`SHOW MASTER STATUS`](/sql-statements/sql-statement-show-master-status.md) | Shows the status of the master server. |
| [`SHOW PLACEMENT`](/sql-statements/sql-statement-show-placement.md) | Shows placement rules. |
| [`SHOW PLACEMENT FOR`](/sql-statements/sql-statement-show-placement-for.md) | Shows placement rules for a specific table. |
| [`SHOW PLACEMENT LABELS`](/sql-statements/sql-statement-show-placement-labels.md) | Shows available placement labels. |
| [`SHOW PLUGINS`](/sql-statements/sql-statement-show-plugins.md) | Shows installed plugins. |
| [`SHOW PRIVILEGES`](/sql-statements/sql-statement-show-privileges.md) | Shows available privileges. |
| [`SHOW PROCESSLIST`](/sql-statements/sql-statement-show-processlist.md) | Shows active threads. |
| [`SHOW PROFILES`](/sql-statements/sql-statement-show-profiles.md) | Shows query profiles. |
| [`SHOW PUMP STATUS`](/sql-statements/sql-statement-show-pump-status.md) | Shows the status of the pump. |
| [`SHOW SCHEMAS`](/sql-statements/sql-statement-show-schemas.md) | Shows all schemas. |
| [`SHOW STATS_HEALTHY`](/sql-statements/sql-statement-show-stats-healthy.md) | Shows the health of statistics. |
| [`SHOW STATS_HISTOGRAMS`](/sql-statements/sql-statement-show-stats-histograms.md) | Shows statistics histograms. |
| [`SHOW STATS_LOCKED`](/sql-statements/sql-statement-show-stats-locked.md) | Shows locked statistics. |
| [`SHOW STATS_META`](/sql-statements/sql-statement-show-stats-meta.md) | Shows metadata statistics. |
| [`SHOW STATUS`](/sql-statements/sql-statement-show-status.md) | Shows server status. |
| [`SHOW TABLE NEXT_ROW_ID`](/sql-statements/sql-statement-show-table-next-rowid.md) | Shows the next row ID for a table. |
| [`SHOW TABLE REGIONS`](/sql-statements/sql-statement-show-table-regions.md) | Shows regions of a table. |
| [`SHOW TABLE STATUS`](/sql-statements/sql-statement-show-table-status.md) | Shows the status of tables. |
| [`SHOW TABLES`](/sql-statements/sql-statement-show-tables.md) | Shows tables in a database. |
| [`SHOW VARIABLES`](/sql-statements/sql-statement-show-variables.md) | Shows system variables. |
| [`SHOW WARNINGS`](/sql-statements/sql-statement-show-warnings.md) | Shows warning messages. |

## Locking statements

| SQL Statement | Description |
|---------------|-------------|
| [`LOCK TABLES`](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md) | Locks tables for the current session. |
| [`UNLOCK TABLES`](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md) | Unlocks tables. |
| [`LOCK STATS`](/sql-statements/sql-statement-lock-stats.md) | Locks statistics for a table. |
| [`UNLOCK STATS`](/sql-statements/sql-statement-unlock-stats.md) | Unlocks statistics for a table. |
```

Reply:
I think this is a great idea and would be very useful.

We should make sure to list things that aren't statements, but are part of statements correctly. See also https://github.com/pingcap/docs/pull/17315

Reply:
yes, the items such as `ADD COLUMN` and `RENAME INDEX` are more like subcommands in the `ALTER TABLE` statement. Perhaps, we can list them as `ALTER TABLE.. ADD COLUMN` and `ALTER TABLE .. RENAME INDEX` on this **SQL Statement Overview** page?

Reply:
I've made these changes in 9b34de6c0c49f3120d336519d54d3ff97fa03760

- Lists SQL statements in groups
- List operations that are part of `ALTER TABLE` as `ALTER TABLE ... <operation>`

The AI generated list had a few nonexistent ones and was missing a few others.

While many people are aware of DML and DDL, I don't think DCL and DQL are very common. So I used different grouping.

I tried to move some of the `SHOW` statements to different groups to keep things together based on what it is managing instead of grouping all `SHOW` statements together.

Reply:
Note that we list:
- `DESC` and `DESCRIBE` (And `EXPLAIN`)
- `SHOW DATABASES` and `SHOW SCHEMAS`

But we don't list:
- `SHOW MASTER STATUS` and `SHOW BINARY LOG STATUS`

```
mysql> desc t1;
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| id    | int(11) | YES  |      | NULL    |       |
+-------+---------+------+------+---------+-------+
1 row in set (0.00 sec)

mysql> describe t1;
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| id    | int(11) | YES  |      | NULL    |       |
+-------+---------+------+------+---------+-------+
1 row in set (0.00 sec)

mysql> explain t1;
+-------+---------+------+------+---------+-------+
| Field | Type    | Null | Key  | Default | Extra |
+-------+---------+------+------+---------+-------+
| id    | int(11) | YES  |      | NULL    |       |
+-------+---------+------+------+---------+-------+
1 row in set (0.00 sec)

mysql> SHOW DATABASES; 
+--------------------+
| Database           |
+--------------------+
| INFORMATION_SCHEMA |
| METRICS_SCHEMA     |
| PERFORMANCE_SCHEMA |
| mysql              |
| sys                |
| test               |
+--------------------+
6 rows in set (0.00 sec)

mysql> SHOW SCHEMAS;
+--------------------+
| Database           |
+--------------------+
| INFORMATION_SCHEMA |
| METRICS_SCHEMA     |
| PERFORMANCE_SCHEMA |
| mysql              |
| sys                |
| test               |
+--------------------+
6 rows in set (0.00 sec)

mysql> SHOW MASTER STATUS;
+-------------+--------------------+--------------+------------------+-------------------+
| File        | Position           | Binlog_Do_DB | Binlog_Ignore_DB | Executed_Gtid_Set |
+-------------+--------------------+--------------+------------------+-------------------+
| tidb-binlog | 449933803186814976 |              |                  |                   |
+-------------+--------------------+--------------+------------------+-------------------+
1 row in set (0.00 sec)

mysql> SHOW BINARY LOG STATUS;
+-------------+--------------------+--------------+------------------+-------------------+
| File        | Position           | Binlog_Do_DB | Binlog_Ignore_DB | Executed_Gtid_Set |
+-------------+--------------------+--------------+------------------+-------------------+
| tidb-binlog | 449933804587450368 |              |                  |                   |
+-------------+--------------------+--------------+------------------+-------------------+
1 row in set (0.00 sec)
```

Not sure what the best way is to list aliases. 

Maybe list both:

| Statement | Description |
|----|---|
| `DESC` | Describe (Alias for `DESCRIBE`) | 
| `DESCRIBE` | Describe | 

Or list them together:

| Statement | Description |
|----|---|
| `DESC` / `DESCRIBE`| Describe| 

Or add a column:

| Statement | Aliases | Description |
|----|---|---|
| `DESCRIBE` | `DESC` | Describe| 

Or don't list them at all and only list them on the statement pages themselves?

Reply:
If we list both, it will be easier for users to find any of them from a table.